### ISSUE TITLE
firebase_auth plugin issue

### ISSUE THREAD

**Issue by shriharip** [flutter/flutter#22903]
java.lang.IllegalArgumentException: Cannot create PhoneAuthCredential without either verificationProof, sessionInfo, ortemprary proof. 

I have followed the exact guidelines mentioned in the document to set up but made only one change. I am not using any other login (anonymous or google). I am just trying to signin with phone number and using the sample phone number and code given in the plugin example. 
I am running on a real device.
I have the SHA1 on my firebase console and also the test phone number and code.

Thanks very much.

**Comment by gildaswise** [comment]
Hey, @shriharip! This probably is just some configuration error. I made [this tutorial](https://medium.com/flutter-community/flutter-adding-sign-in-with-google-and-phone-authentication-to-your-app-69f681518f9b) a while ago on how to setup `signInWithPhoneNumber` (and also Google authentication, but that's not the focus of the tutorial), could you try to follow it and see if the error persists?

**Comment by shriharip** [comment]
@gildaswise Hi, yeah that was a nice tutorial. I had checked that and some parts of my code and both were similar. I saw that you did the google signIn first and then the phone verification for me it was just the phone verification. Hence did not go through that in detail.
Let me try your repo and removing google signin.

thanks

**Comment by shriharip** [comment]
@gildaswise . I cloned your repo and removed social login(google) and used my real phone number and i did get a code and authentication was successful.

I tried to use a test number that is used in the firebase_auth plugin example and the code 888888 and it was working :)

I copied my gradle.build and app/gradle.build and my service.json file so I don't know what is the problem with the example app. but yours is working fine and I will close this since it has solved my issue.

thanks,
Hari
